Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean

```diff
@@ -169,6 +171,36 @@ instance : Decidable G.Connected := by
 instance instDecidableMemSupp (c : G.ConnectedComponent) (v : V) : Decidable (v ∈ c.supp) :=
   c.recOn (fun w ↦ decidable_of_iff (G.Reachable v w) $ by simp)
     (fun _ _ _ _ ↦ Subsingleton.elim _ _)
+
+lemma odd_card_odd_components (ho : Odd (Nat.card V)) :
+    Odd (Nat.card ({(c : ConnectedComponent G) | Odd (Nat.card c.supp)})) := by
```

Comment:
Can you also turn this into an `iff`, and golf its proof? (e.g. the lemma `odd_sum_iff_odd_card_odd` should make this lemma shorter, right?)

Reply:
Done:) I suspect someone with more experience could golf it a bit more, but this is as far as I'm able to take it currently.